sync/src/chain.rs

```diff
@@ -658,6 +668,10 @@ impl ChainSync {
 			return Ok(());
 		}
 
+		if self.sync_start_time.is_none() {
```

Comment:
Moved that after the initial checks, otherwise in case we are not using warp-only sync, we might timeout a bit too early (I actually got a lot of peers on different chains from discovery), without even giving a chance to start warping.